packages/sanity/src/config/prepareConfig.ts

```diff
@@ -216,15 +217,51 @@ function resolveSource({
   const {dataset, projectId} = config
   const bifur = getBifurClient(client, auth)
   const errors: unknown[] = []
+  const clients: Record<string, SanityClient> = {}
+  const getClient = (options: SourceClientOptions): SanityClient => {
+    if (!options || !options.apiVersion) {
+      throw new Error('Missing required `apiVersion` option')
+    }
+
+    if (!clients[options.apiVersion]) {
+      clients[options.apiVersion] = client.withConfig(options)
+    }
+
+    return clients[options.apiVersion]
+  }
 
   const context = {
     client,
+    getClient,
     currentUser,
     dataset,
     projectId,
     schema,
   }
 
+  // <TEMPORARY UGLY HACK TO PRINT DEPRECATION WARNINGS ON USE>
+  /* eslint-disable no-proto */
+  const wrappedClient = client as any
```

Comment:
Possibly extract/reuse code. But its getting deleted soon so I guess it doesnt matter.